src/drivers/smmu/smmuv2.c

```diff
@@ -547,13 +547,13 @@ void smmu_sid_bind_cb(word_t sid, word_t cb)
     reg = S2CR_PRIVCFG_SET(S2CR_PRIVCFG_DEFAULT);
     reg |= S2CR_TYPE_SET(S2CR_TYPE_CB);
     reg |= S2CR_CBNDX_SET(cb);
-    smmu_write_reg32(SMMU_GR0_PPTR, SMMU_S2CRn(sid), reg);
+    smmu_write_reg32(SMMU_GR0_PPTR, SMMU_S2CRn(cb), reg);
```

Comment:
I don't understand this change or the commit message, can you please elaborate? Surely it is not arbitrary whether this should be the `sid` or the `cb` here.

Reply:
Looking back at this piece of the SMMUv2 driver, I think sid is probably the correct value to use
here for the TX2 platform, however, this causes issues on a zynqmp platform.

The TX2 platform implements 128 stream mapping groups and 7 bit stream ids therefore a max value
of 128 so using the sid here in that case is not a problem and probably the preferred method.

The zynqmp, however, implements only 48 stream mapping groups and allows stream ids all the way up 
to a max value of 32767. Therefore, using the sid value here will quickly overrun the stream mapping 
register space and start writing data to unintended locations.

In this current SMMUv2 implementation with the constraint of 1 stream id per context bank using the
context bank index here is more appropriate for the zynqmp, however, this would need to change to provide a smarter mapping of sid value to one of the 48 mapping groups in the future to support multiple stream ids per context bank.

All of that is to say, I think the following changes should be made to this commit:
 - Wrap these changes in a platform ifdef. One that uses the original method with the sid for the TX2 and one that uses the cb for the zynqmp.
 - Update the commit message to better describe the changes
 - Move this commit to the zynqmp platform specific PR seL4/seL4#761
 
 Do those changes seem appropriate?